controllers/space/space_controller.go

```diff
@@ -422,13 +422,15 @@ func NewNSTemplateSetSpec(space *toolchainv1alpha1.Space, bindings []toolchainv1
 	}
 	if tmplTier.Spec.ClusterResources != nil {
 		s.ClusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
-			TemplateRef: tmplTier.Spec.ClusterResources.TemplateRef,
+			TemplateRef: tmplTier.Status.Revisions[tmplTier.Spec.ClusterResources.TemplateRef],
 		}
 	}
 	if len(tmplTier.Spec.Namespaces) > 0 {
 		s.Namespaces = make([]toolchainv1alpha1.NSTemplateSetNamespace, len(tmplTier.Spec.Namespaces))
 		for i, ns := range tmplTier.Spec.Namespaces {
-			s.Namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace(ns)
+			s.Namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace{
+				TemplateRef: tmplTier.Status.Revisions[ns.TemplateRef],
+			}
```

Comment:
we should make sure that we start using the refs from the revisions map only when the map contains references for all TierTemplates from the Spec. 
There can be still some small time gap when the NSTemplateTier.Spec is updated, but the status part doesn't contain the refs for all TierTemplates - in these cases, the refs in Namespaces would contain an empty string, which could trigger the deletion of the already existing namespace.
Let's add a simple "validate" function which validates that the revisions map contains refs for all TierTemplates, if not, then the controller should return an error

Reply:
Done , PTAL